### DISCUSSION TITLE
Provider in wrapPage to wrapRoot

### DISCUSSION THREAD

**Discussion by GuilhemBarboux** [gatsbyjs/themes#122] [Ideas]
Hi there,

Do you think it's a good idea to move to move the provider on the root wrapper ? because actually we can't use locale on wrapPage, and if you want to keep elements between page (like loading or transition) we can't translate their.

Thank you

**Comment by tx0c** [comment]
yes, it's a very big issue, not only a question under discussion

https://github.com/gatsbyjs/themes/issues/138#issuecomment-949082541
